contributing/code/bc.rst

```diff
@@ -277,6 +277,12 @@ Change value of a constant                          Yes [1]_ [5]_
        Additionally, if a constant will likely be used in objects that are
        serialized, the value of a constant should not be changed.
 
+.. [6] Allowed using the ``@final`` annotation.
+
+.. [7] Allowed if the class is final.
```

Comment:
this doc is here do make things explicit, I think even if it's clear for us, I'm sure it's not for someone not used to our processes. since we always add the version were it was made final, I don't see the issue (except the "wording" one of course, but that's solvable :)